dev/sparktestsupport/toposort.py

```diff
@@ -0,0 +1,85 @@
+#######################################################################
+# Implements a topological sort algorithm.
```

Comment:
Yes just append. I don't know if we've got a strong format going, except to try to precede the text from Foo with a line like "Foo" or "For Foo:". Anything sane should be OK.

Technically you reproduce all relevant parts of the `NOTICE` and only relevant parts. All is relevant here. Really, not sure the project needed this file but hey. Practically you could argue you can abbreviate it, but given it's short, I'd just copy it.